samtranslator/swagger/swagger.py

```diff
@@ -852,6 +852,8 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
         """
         if resource_policy is None:
             return
+        if not isinstance(resource_policy, dict):
+            raise InvalidDocumentException([InvalidTemplateException("Resource Policy is not a valid dictionary.")])
```

Comment:
LGTM, this is the fix and others are just changing the exception types right?

Reply:
+1 on this question. Also, not sure if resource_policy here is allowed to be an instrinsic expression, like `!Ref: sth`. If so, maybe we need to consider this.

Reply:
that and adding tests so to not regress in future.

Reply:
@wchengru I am not sure sure what you mean. Can you explain further?

Reply:
As per @wchengru 's suggestion tried a template with resourcepolicy: !ref but did not work. We probably do not support it here.